src/site/content/en/blog/granular-chunking-nextjs/index.md

```diff
@@ -0,0 +1,273 @@
+allows for multiple requests to be streamed in parallel using a single connection over a single
+origin. In other words, we generally do not need to worry about limiting the number of chunks
+emitted by our bundler.
+
+[All major browsers](https://caniuse.com/#feat=http2) support HTTP/2, and we wanted to see for
+ourselves if increasing the number of requests by splitting Next.js's single "commons" bundle into
+multiple shared chunks would affect loading performance in any way. We began by measuring the
+performance of a single site while modifying the maximum number of parallel requests using the
+[`maxInitialRequests`](https://webpack.js.org/plugins/split-chunks-plugin/#splitchunksmaxinitialrequests) property.
+
+![Page load performance with increased number of requests](./performance-num-requests-1.png)
+
+In an average of three runs of multiple trials on a single web page, the load, start-render and
+first contentful paint times all remained about the same when varying the max initial request count
+(from 5 to 15). Interestingly enough, we noticed a slight performance overhead only after splitting
+aggressively to hundreds of requests.
+
+![Page load performance with hundreds of requests](./performance-num-requests-2.png)
+
+This showed that staying under a reliable threshold (20~25 requests) struck the right balance
+between loading performance and caching efficiency. After some baseline testing, 25 was selected as
+the `maxInitialRequest` count. 
+
+Modifying the maximum number of requests that happen in parallel resulted in more than a single
+shared bundle, and separating them appropriately for each entry point significantly reduced the
+amount of unneeded code for the same page.
+
+![JavaScript payload reductions with increased chunking](./js-payload-num-requests.png)
+
+## Size reductions with granular chunks
+
+Many frameworks, including Next.js, rely on client-side routing (handled by JavaScript) to inject
+newer script tags for every route transition. But how do we predetermine these dynamic chunks at build time?
+
+Next.js uses a server-side build manifest file to determine which outputted chunks are used by
+different entry points. To provide this information to the client as well, an abridged client-side
+build manifest file was created to map all the dependencies for every entry point.
+
+```javascript
+// Returns a promise for the dependencies for a particular route
+getDependencies (route) {
+  return this.promisedBuildManifest.then(
+    man => (man[route] && man[route].map(url => `/_next/${url}`)) || []
+  )
+}
+```
+
+![Output of multiple shared chunks in a Next.js application](./outputted-chunks.png)
+
+This newer granular chunking strategy was first rolled out in Next.js behind a flag, where it was tested on a
+number of early adopters. Many saw significant reductions to the total JavaScript used for their
+entire site:
+
+<div class="w-table-wrapper">
```

Comment:
No page load performance data?

Reply:
We've been focusing on total JS reductions (since for many of the sites, the JS reduced on only the home page may not be substantial enough to affect single load page performance). The benefits are mostly noticed during page transitions (less JS fetched as the user navigates throughout the app).

I'll try to source some numbers for this however.